### DISCUSSION TITLE
"@container": "@annotation"

### DISCUSSION THREAD

**Discussion by Tpt** [json-ld/json-ld-star#12] [Ideas]
Thank you for this work on JSON-LD-star. It's great to see it coming forward!

I encountered multiple datasources where an attribute I would like to parse as an RDF* annotation is in the same level as the other attributes of the triple object.

For example I would like to be able to parse `"role"` into an  annotation of the `"actor"` relation from the following JSON:
```json
{
   "name": "My movie",
   "actor": {
        "name": "John Doe",
        "role": "the vilain"
   }
}
```

for that, allowing something like  `"@container": "@annotation"` in the context might be convenient.

To follow the example, it might be very handy to write a context like:
```json
{
   "@vocab": "http://schema.org/",
   "actor": {"@type": "Person"},
   "role": {"@container": "@annotation"}
}
```

To get the following RDF*:
```turtle
@prefix schema: <http://schema.org/>

_:a schema:name "My movie" .
_:a schema:actor _:b .
<< _:a schema:actor _:b . >> schema:role "the vilain" .
_b: a schema:Actor .
_b: schema:name "John Doe" .
```

What do you think about it?

**Comment by pchampin** [comment]
@Tpt if you have any reference to such formats, we would be interested (to document the existing use-cases).

If this is a common practice, I agree that we should address it.
I sympathize with the will to reuse an existing mechanism, but I am not entirely sure about `@container` for this. The general effect of container is to alter the *value* of the property, by wrapping it (e.g. `@list`, `@graph`) or unwrapping it (language-map, id-map, and other *-maps). What you suggest here is to wrap *the property and the value* into something that is added to the parent object.
This is just a gut-feeling, I'm not either strongly against it.

Possible alternative: use `"@annotation": true` in the term definition?

**Reply by Tpt** [reply]
> @Tpt if you have any reference to such formats, we would be interested (to document the existing use-cases).

It's a great question. I don't have any good example of public formats doing that. My example is taken from a proprietary one.

**Reply by gkellogg** [reply]
It could be a question of what creates the least intellectual load; `@container: @annotation` might be clear. Adding a new `@annotation: true` to a term definition does clarify its use.

Either could allow the use within a value object, which otherwise could not contain any other properties.

(It would be nice if a conversation could be turned into an issue, as there are too many places to remember to look as it is).
